mc-sema/cfgToLLVM/JumpTables.cpp

```diff
@@ -342,6 +342,57 @@ static void doJumpTableViaSwitchReg(
 
 }
 
+void doJumpOffsetTableViaSwitchReg(
+        BasicBlock *& block, 
+        InstPtr ip, 
+        Value *regVal,
+        BasicBlock *&default_block,
+        Value *data_location,
+        MCSOffsetTablePtr ot_ptr)
+{
+
+    llvm::errs() << __FUNCTION__ << ": Doing jump offset table via switch(reg)\n";
+    Function *F = block->getParent();
+    
+
+    // create a default block that just traps
+    default_block = 
+        BasicBlock::Create(block->getContext(), "", block->getParent(), 0);
+    // end default block
+
+    const std::vector< std::pair<VA,VA> > &offset_dest = ot_ptr->getConstTable();
+    std::unordered_map<VA,VA> uniq_blocks;
```

Comment:
Wouldn't LLVM's SwitchInst generation automatically order the blocks how it sees fit? After all, its a Constant->Function mapping, it shouldn't need to preserve the order in which you add the cases. Or is that a thing that SwitchInst promises? Have not read the docs that thoroughly